src/components/fields/error-field/types.ts

```diff
@@ -0,0 +1,13 @@
+import { IYupValidationRule } from "../../frontend-engine/types";
+import { IBaseFieldSchema } from "../types";
+
+export interface IErrorFieldValidationRule extends IYupValidationRule {
+	/** for customising error message when field is present, not visible to user */
+	error?: boolean | undefined;
+}
+
+export interface IErrorFieldSchema
+	extends Pick<
+		IBaseFieldSchema<"error-field", undefined, IErrorFieldValidationRule>,
+		"showIf" | "uiType" | "validation"
+	> {}
```

Comment:
this field is special, all other validations don't have effect here, I propose we narrow the `validation` typing further to allow the `error` rule only

```suggestion
export interface IErrorFieldValidationRule {
	/** for customising error message when field is present, not visible to user */
	error: boolean;
	errorMessage: string;
}

export interface IErrorFieldSchema extends Pick<IBaseFieldSchema<"error-field">, "showIf" | "uiType"> {
	validation?: [IErrorFieldValidationRule];
}
```

Reply:
can we also allow the field to have `children`? this way any messages can be rendered within it and we can save the effort to do conditionally rendering on 2 components